pkg/objcache/objcache.go

```diff
@@ -112,9 +100,32 @@ func (c *Cache) Create(key string, size int64) (writer io.Writer, err error) {
 			return nil, ErrCacheFull
 		}
 	}
-	c.entries[key] = NewBuffer(make([]byte, 0, int(size)))
-	c.currentSize += valueLen
-	return c.entries[key], nil
+
+	// Will hold the object contents.
+	buf := bytes.NewBuffer(make([]byte, 0, size))
+	// Account for the memory allocated above.
+	c.currentSize += uint64(size)
+	// Implements io.WriteCloser
+	return struct {
```

Comment:
Write this more cleanly do not use an anonymous function, implement a proper struct and fill it with io.Writer and Closer.